### PR TITLE
Handles different return types.

### PR DESCRIPTION
**Pull request by AndrewMcBurney**
#3, #41

### Changes
- Adds ability to return different types in functions.
- Adds a `ReturnStatement` class for returning expressions.

### Usage

```js
// Declare reference
const zero = new cgl.Reference(
    new cgl.InterfaceVariable(
        cgl.Qualifier.In, new cgl.VariableSource(new cgl.Type(cgl.Kind.Int), '0')
    )
);

/**
 * int returnZero() {
 *   return 0;
 * }
 */
new cgl.Function('returnZero', [new cgl.ReturnStatement(zero)], cgl.Kind.Int);
```

### CODE REVIEW

File: src/return_statement.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import InterfaceVariable from './interfacevariable';
+import Set from './util/set';
+import SyntaxNode from './syntaxnode';
+
+export default class ReturnStatement implements SyntaxNode {
```
Comment (pbardea) [inline src/return_statement.ts]: Did we want to consider adding a type for the return statement? This way we could type check that the type matches the function return type. We would do that later after the AST is constructed.

Comment (AndrewMcBurney) [reply src/return_statement.ts]: Yeah, sounds good to me 👍

File: src/return_statement.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import InterfaceVariable from './interfacevariable';
+import Kind from './kind';
+import Set from './util/set';
+import SyntaxNode from './syntaxnode';
+
+export default class ReturnStatement implements SyntaxNode {
+    private node: SyntaxNode;
+    public readonly returnType: Kind;
+
+    constructor(node: SyntaxNode, returnType: Kind) {
```
Comment (abhimadan) [inline src/return_statement.ts]: I think `node` should be an expression instead; then we know the `ReturnStatement`'s type is `expr.returnType()`.

Comment (AndrewMcBurney) [reply src/return_statement.ts]: That's a good point, will make the changes! 👍

**Pull request merged**